src/numeric-types.c

```diff
@@ -93,6 +83,102 @@ numeric_##type##_to_##gtype (numeric_##type num)                     \
     return data.v_##gtype;                                           \
 }
 
+typedef struct _NumericTypeInfo
+{
+    GType             type;
+    const gchar      *name;
+    gsize             width;
+    NumericByteOrder  byte_order;
+} NumericTypeInfo;
+
+GArray *TYPE_TABLE = NULL;
+
+void
+numeric_type_register_static (GType        type,
+                              const gchar *name,
+                              gsize width,
+                              NumericByteOrder byte_order)
```

Comment:
The alternative is to expose the `NumericTypeInfo` structure so we can use a single argument for the metadata.